db/migrate/20200608163713_create_storage_system_types.rb

```diff
@@ -0,0 +1,11 @@
+class CreateStorageSystemTypes < ActiveRecord::Migration[5.1]
+  def change
+    create_table :storage_system_types do |t|
```

Comment:
See https://github.com/ManageIQ/manageiq-schema/pull/505/files#r482449221 - I'm concerned this name might create confusion, and I wonder if we can come up with a different name.  @agrare Do we have a precedent for table names ending in `_types`?

Reply:
@Fryguy we have a few `cloud_volume_types`, `miq_worker_types`, `notification_types`, and `pxe_image_types`

Reply:
Ok, then I guess it's ok. :+1:

Reply:
That said you're right another model referencing this does look like one half of a polymorphic association so if we can come up with a better name I'm all for it

Reply:
@gregoryb-at-ibm what would populate this storage_system_types and what are some of the possible values?  Would this be e.g. SVC or V7000?  In that case this can probably just be an STI class.  Or is it something collected from the provider API that is very dynamic and has to be populated by refresh?

Reply:
@agrare , The storage_system_types are supposed to be scanned from whatever backend the EMS is managing. It's up to the EMS and the backend to determine what to put there, based on how they want to differentiate different storage systems.
So one backend could have a list like ["SVC", "V7000", "Netapp"], and that can be enough for them.
Another backend could need to use advanced functions of SVC which would require to differentiate between different SVC versions, their storage_system_types could look like ["svc_version_lte_x", "svc_version_gt_x", "V8K"]. 
Note that the lists don't adhere to the same naming conventions because they use conventions from the EMS's backend.

They are used when creating a storage system. The user needs to pick one from a drop-down, since the EMS backend needs to know what kind of a system it is in order to connect to it. 
They are also presented to the user when viewing systems.
In the future we want to use them to determine what features the system supports, and perhaps add uniformity across EMSs.

Reply:
@agrare , Can this conversation be resolved? I've renamed storage_system_types to storage_system_families so it doesn't look like a polymorphic relationship.

Reply:
:+1: